backends/tc/backend.cpp

```diff
@@ -912,4 +957,12 @@ void ConvertToBackendIR::updateMatchType(const IR::P4Table *t, IR::TCTable *tabl
     tabledef->setMatchType(tableMatchType);
 }
 
+bool ConvertToBackendIR::checkParameterDirection(const IR::TCAction *tcAction) {
+    bool dirParam = false;
+    for (auto actionParam : tcAction->actionParams) {
+        if (actionParam->getDirection() != TC::NONE) dirParam = true;
```

Comment:
optimize to break after 1st direction type found

Reply:
Updated the code.